### ISSUE TITLE
Telegraf should use the couchbase official SDK

### ISSUE THREAD

**Issue by sparrc** [influxdata/telegraf#2418]
The client library we're currently using is not the official Couchbase library

we should change to use the new official 1.0 version: https://github.com/couchbase/gocb

see also: https://blog.couchbase.com/go-sdk-1.0-ga/

**Comment by danielnelson** [comment]
I think we are using functionality from the `go-couchbase` library that does not seem to have a replacement in the `gocb` library.  In particular there does not seem to be a way to call the `/pools/default` or `/pools/default/buckets/<bucket>/stats` endpoints.

I looked into using the Stats method on a Bucket, but this provides lower level stats similar to what you get from cbstats.  I'm not sure how we can recreate the current stats from these stats.

Here are the best docs I can find for the stats, not all of them are returned by my version of couchbase:
https://developer.couchbase.com/documentation/server/4.6/cli/cbstats-intro.html

How do you think we should move forward?

**Comment by sparrc** [comment]
might be best to open an issue on the https://github.com/couchbase/gocb repo asking for an equivalency, and see what they say.

I would also ask which metrics linked above are available via the Go client lib.

Unfortunately it might just be that the client library doesn't provide very good stats support, and we may just need to figure out which endpoints we should be querying. I can see that couchbase has some sort of stat collecting python agent here: https://github.com/couchbase/cbagent

That being said, that would probably be a larger project and not suitable for the 1.3 release.

**Comment by ingenthr** [comment]
I was chatting with @danielnelson on this over on gitter.  Based on the discussion there, @daschl suggested that we add to gocb a REST generic but topology aware interface to gocb which would let you get the info you're looking for.  I looked at the stats he'd pointed to, and it seems to me this would work well.  It'll be best to get the stats from the cluster's REST interface, as that way any aggregations are consistent with what the cluster itself is reporting and as new stats are added they'll be immediately available to telegraf.

As of this writing, @brett19 is on vacation for a couple days but we can pick this up when he's back and find the right way forward.  For tracking in gocb, I've filed https://issues.couchbase.com/browse/GOCBC-164
